src/mnode/ticket-processor.cpp

```diff
@@ -261,19 +275,34 @@ bool CPastelTicketProcessor::ValidateIfTicketTransaction(const int nHeight, cons
             //in this tickets last 2 outputs is: change, and payment to the seller
             if (ticket_id == TicketID::Trade)
             {
-                if (i == num - 2)
-                    continue;
-                if (i == num - 1)
-                {
-                    if (tradePrice != tx.vout[i].nValue)
-                    {
-                        bOk = false;
-                        error_ret = strprintf("Wrong payment to the seller: expected - %d, real - %d", tradePrice, tx.vout[i].nValue);
-                        break;
-                    }
-                    continue;
+                if (i == (hasRoyaltyFee && hasGreenFee ? num - 4 :
+                          hasRoyaltyFee || hasGreenFee ? num - 3 : num - 2)) {
+                  continue;
+                } else if (i == (hasRoyaltyFee && hasGreenFee ? num - 3 :
+                                 hasRoyaltyFee || hasGreenFee ? num - 2 : num - 1)) {
+                  if (tradePrice != tx.vout[i].nValue) {
+                    bOk = false;
+                    error_ret = strprintf("Wrong payment to the seller: expected - %d, real - %d", tradePrice, tx.vout[i].nValue);
```

Comment:
just noticed - tradePrice and tx.vout[i].nValue is int64_t, so instead of %d we have to use PRId64

Reply:
ok